### ISSUE TITLE
Leaks

### ISSUE THREAD

**Issue by chaserCN** [RxSwiftCommunity/Action#41]
Hi.

I have accidentally googled this issue in RxSwift: https://github.com/ReactiveX/RxSwift/issues/573
and decided to check CocoaActions.

```
   class ReleaseMe {
        var name: String

        init(name: String) {
            self.name = name
        }

        deinit {
            print("deinit for \(name)")
        }
    }

    func request() -> Observable<Void> {
        let captured = ReleaseMe(name: "captured")

        return Observable.just(ReleaseMe(name: "just"))
            .map {_ in
                captured.name
            }
            .map(void)
    }

    private func bindModel() {
        loginButton.rx_action = CocoaAction() {[unowned self] in
            return self.request()
        }
    }

```

ReleaseMe objects are not freed when i tap the button. 
I managed to make it work only after i got rid of ReplaySubject and addDisposableTo in Action class, but i guess it is not an option :)

**Comment by ashfurrow** [comment]
Cool, thanks for reporting the issue. I'll take a look asap.

**Comment by ashfurrow** [comment]
I'm curious, if you tap the button a second time, does the first `ReleaseMe` get released?

**Comment by werediver** [comment]
Memory management issues are what I always consider something serious.

@ashfurrow It's not clear from your last comment whether you've been able to reproduce the issue or it seems to be all right there?

**Comment by werediver** [comment]
Is it related to the pull-request #33?

**Comment by eXhausted** [comment]
any updates?

**Comment by xnxin** [comment]
Is this problem fixed?

**Comment by ashfurrow** [comment]
It sounds like #33 fixed this but I can't remember for sure. If someone were able to test for us, that would be a welcome contribution 🙇

**Comment by ixrevo** [comment]
Action 3.1.1 
Cannot reproduce this.

**Comment by ashfurrow** [comment]
Thanks for checking @ixrevo 🙇 I'm going to close this, but if anyone finds any problems please don't hesitate to let us know in a new issue.
